### ISSUE TITLE
Faction bases

### ISSUE THREAD

**Issue by Sirbab** [CleverRaven/Cataclysm-DDA#39070]
# Describe the bug

When building some faction base expansions that can use different materials, an in progress expansion does not hide the other expansions of the same location.

# Steps To Reproduce

<!-- Steps to reproduce the behavior:
1. Make a faction base.
2. Expand it until you have a factory expansion.
3. Send an NPC to build a metal expansion north east.
4: See that metal northeast expansion is missing, but that you can send an NPC to build wood, stone, earth etc for northeast wall even though metal is already in progress.

# Expected behavior

I expect that sending an NPC to construct an expansion area should be mutually incompatible with other material types of the same area, and hide them.

# Screenshots

![image](https://user-images.githubusercontent.com/1490991/77794214-d9f33000-7041-11ea-9f52-e64b7d98740e.png)

# Versions and configuration

<!-- Please complete the following information; you can now go to:
  - OS: Windows
    - OS Version: 10.0 1909
- Game Version: 0.D-13428-g4476c72 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Aftershock [aftershock],
    Magiclysm [magiclysm],
    Dorf Life [Dorf_Life],
    SpeedyDex [speedydex],
    Mutation Changes [Mutation_Changes],
    Stats Through Skills [StatsThroughSkills]
]

# Additional context
N/A

**Comment by Sirbab** [comment]
Apologies if the formatting is kind of off, or if i missed something.
This is my first issue report with cata.

**Comment by Maleclypse** [comment]
So right now, if you were building a livestock upgrade you could make some barns out of wood and some out of stone.  Are you saying you have the ability to rebuild over the the already built buildings with different materials or are you saying if you build one building with stone you should have to build all buildings with stone?  Or do you have half a building made of stone and half of wood?

**Comment by Sirbab** [comment]
I haven't actually tried to send my workers out to build multiple iterations of the same one at once.  I can check once i progress far enough if it hides them once it's done.
What i mean is that if you start to build the Northwest side of a building, you shouldn't be able to send a worker to build *In that same exact spot* in a different material as that would be building one on top of the other.

Obviously if you build one half the building out of wood, you should be able to build the other have out of stone, just not in the same spot at the same time.

**Comment by Maleclypse** [comment]
If I’m reading your screenshot correctly the other material versions are lowlit and can’t be selected to build unless you recall the guy currently working before he finishes. I believe it won’t actually hide them until construction is complete. In case you decide you picked the wrong material.

**Comment by Sirbab** [comment]
Nah, they're low lit because i don't fulfill the other material requirements.  They were lit up when I i satisfied the other requirements.
![image](https://user-images.githubusercontent.com/1490991/77815420-4f3a2180-7091-11ea-9fb7-cf3f328a58d5.png)
See?
